appservice/src/SiteWrapper.ts

```diff
@@ -116,6 +126,61 @@ export class SiteWrapper {
         this.log(outputChannel, 'Deployment completed.');
     }
 
+    public async localGitDeploy(fsPath: string, client: WebSiteManagementClient, outputChannel: vscode.OutputChannel, servicePlan: string): Promise<string | undefined> {
+        const kuduClient: KuduClient = await this.getKuduClient(client);
+        const yes: string = 'Yes';
+        const pushReject: string = localize('localGitPush', 'Push rejected due to Git history diverging. Force push?');
+        const success: string = 'success';
+        const scmType: string = 'LocalGit';
+
+        const [publishCredentials, config]: [User, SiteConfigResource] = await Promise.all([
+            this.getWebAppPublishCredential(client),
+            this.getSiteConfig(client)
+        ]);
+
+        if (config.scmType !== scmType) {
+            // SCM must be set to LocalGit prior to deployment
+            const scmUpdate: string | undefined = await this.updateScmType(client, config, scmType);
+            if (!scmUpdate) {
+                return undefined;
+            }
+        }
+        // credentials for accessing Azure Remote Repo
+        const username: string = publishCredentials.publishingUserName;
+        const password: string = publishCredentials.publishingPassword;
+        const remote: string = `https://${username}:${password}@${this._gitUrl}`;
+        const localGit: git.SimpleGit = git(fsPath);
+        try {
+
+            const status: git.StatusResult = await localGit.status();
+            if (status.files.length > 0) {
+                const uncommit: string = localize('localGitUncommit', `${status.files.length} uncommitted change(s) in local repo "${fsPath}"`);
+                vscode.window.showWarningMessage(uncommit);
+            }
+            await localGit.push(remote, 'HEAD:master');
+        } catch (err) {
+            // tslint:disable-next-line:no-unsafe-any
+            if (err.message.indexOf('spawn git ENOENT') >= 0) {
+                throw new errors.GitNotInstalledError();
+            } else if (err.message.indexOf('error: failed to push') >= 0) { // tslint:disable-line:no-unsafe-any
+                const input: string | undefined = await vscode.window.showErrorMessage(pushReject, yes);
+                if (input === 'Yes') {
+                    await (<(remote: string, branch: string, options: object) => Promise<void>>localGit.push)(remote, 'HEAD:master', { '-f': true });
+                    // Ugly casting neccessary due to bug in simple-git. Issue filed:
+                    // https://github.com/steveukx/git-js/issues/218
+                } else {
+                    return undefined;
+                }
+            } else {
+                // tslint:disable-next-line:no-unsafe-any
+                throw new errors.LocalGitDeployError(err, servicePlan);
+            }
+        }
+
+        await this.waitForDeploymentToComplete(kuduClient, outputChannel);
+        return success;
```

Comment:
I would change waitForDeploymentToComplete to return DeployResult and just do:
```
return await this.waitForDeploymentToComplete(kuduClient, outputChannel);
```

 (rather than defining your own string for success)